src/settings/mod.rs

```diff
@@ -119,36 +120,51 @@ pub struct Settings {
     pub pyupgrade: pyupgrade::settings::Settings,
 }
 
-static DEFAULT_EXCLUDE: Lazy<Vec<FilePattern>> = Lazy::new(|| {
-    vec![
-        FilePattern::Builtin(".bzr"),
-        FilePattern::Builtin(".direnv"),
-        FilePattern::Builtin(".eggs"),
-        FilePattern::Builtin(".git"),
-        FilePattern::Builtin(".hg"),
-        FilePattern::Builtin(".mypy_cache"),
-        FilePattern::Builtin(".nox"),
-        FilePattern::Builtin(".pants.d"),
-        FilePattern::Builtin(".ruff_cache"),
-        FilePattern::Builtin(".svn"),
-        FilePattern::Builtin(".tox"),
-        FilePattern::Builtin(".venv"),
-        FilePattern::Builtin("__pypackages__"),
-        FilePattern::Builtin("_build"),
-        FilePattern::Builtin("buck-out"),
-        FilePattern::Builtin("build"),
-        FilePattern::Builtin("dist"),
-        FilePattern::Builtin("node_modules"),
-        FilePattern::Builtin("venv"),
-    ]
-});
-
-static DEFAULT_DUMMY_VARIABLE_RGX: Lazy<Regex> =
-    Lazy::new(|| Regex::new("^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$").unwrap());
-
 impl Settings {
     pub fn from_configuration(config: Configuration, project_root: &Path) -> Result<Self> {
+        let mut rules = RuleTable::empty();
+
+        let fixable = resolve_codes([RuleCodeSpec {
+            select: &config.fixable.unwrap_or_else(|| CATEGORIES.to_vec()),
+            ignore: &config.unfixable.unwrap_or_default(),
+        }]);
+
+        for code in validate_enabled(resolve_codes(
+            [RuleCodeSpec {
+                select: &config.select.unwrap_or_else(|| defaults::PREFIXES.to_vec()),
+                ignore: &config.ignore.unwrap_or_default(),
+            }]
+            .into_iter()
+            .chain(
+                config
+                    .extend_select
+                    .iter()
+                    .zip(config.extend_ignore.iter())
+                    .map(|(select, ignore)| RuleCodeSpec { select, ignore }),
+            )
+            .chain(
+                // If a docstring convention is specified, force-disable any incompatible error
+                // codes.
+                if let Some(convention) = config
+                    .pydocstyle
+                    .as_ref()
+                    .and_then(|pydocstyle| pydocstyle.convention)
+                {
+                    Left(iter::once(RuleCodeSpec {
+                        select: &[],
+                        ignore: convention.codes(),
+                    }))
+                } else {
+                    Right(iter::empty())
+                },
+            ),
+        )) {
+            let fix = fixable.contains(&code);
+            rules.enable(code, fix);
+        }
```

Comment:
Can we move this into its own utility function, to keep the constructor declarative?

Reply:
We could however I don't really see the point since 1) that function won't be used anywhere else and 2) the function signature would have to be quite complex since it would need to take `config.fixable`, `config.unfixable`, `config.select` and `config.ignore` (we cannot just pass `&Configuration` because we want to avoid needless clones() and so we'd need to have 4 owned parameters for the borrow checker).

Reply:
I think the point would be to increase the readability of this constructor :)

Reply:
Fair enough, done :)

Reply:
Thank you, will merge shortly :)